### PR TITLE
[CSL-2100] Add Support for Browse Groups

### PR DESCRIPTION
**Pull request by Mudaafi**

### CODE REVIEW

File: AutocompleteClient/FW/Logic/Result/CIOFilterGroup.swift [hunk 1]
```diff
@@ -54,11 +59,13 @@ public class CIOFilterGroup: NSObject {
...
         let parents: [CIOFilterGroup] = parentObj?.compactMap { obj in return CIOFilterGroup(json: obj) } ?? []
         let count = json["count"] as? Int
+        let data = json["data"] as? JSONObject
 
         self.displayName = name
         self.groupID = groupID
         self.count = count
         self.children = children
         self.parents = parents
+        self.data = data
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIOFilterGroup.swift:69]: Nice!

**Review by jjl014 — COMMENTED** [review]
Nice work and thanks for being thorough with the tests!

I did leave some comments abstracting certain things a bit and simplifying some of the request object variables. Let me know what you think. Thanks!

File: AutocompleteClient/FW/Logic/Request/Builder/CIOBrowseGroupsQueryBuilder.swift [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+/**
+ Request builder for creating an browse groups query.
+ */
+public class CIOBrowseGroupsQueryBuilder {
+    
+    /**
+     The filters used to refine results
+     filters[group_id] - The id of the specific group that should be included in the response
+     */
+    var filters: CIOQueryFilters?
```
[the remark this answers is not in the document]
Comment (Mudaafi) [reply AutocompleteClient/FW/Logic/Request/Builder/CIOBrowseGroupsQueryBuilder.swift:20]: Totally cool and would prefer simplifying the objects, but wasn't sure if we use/will need to use other filters/fmt_options in this query. Might not be "future-proof" in that sense but if it's a non-issue currently, I can refactor it so we only consider the group_id & max_depth.

File: AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift [hunk 3]
```diff
@@ -174,6 +174,11 @@ public class ConstructorIO: CIOSessionManagerDelegate {
         let request = self.buildRequest(data: query)
         executeBrowse(request, completionHandler: completionHandler)
     }
+    
+    public func browseGroups(forQuery query: CIOBrowseGroupsQuery, completionHandler: @escaping BrowseQueryCompletionHandler) {
```
[the remark this answers is not in the document]
Comment (Mudaafi) [reply AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift:178]: Will do!

**Review by stanlp1 — APPROVED** [review]
This is looking good to me.

**Comment by jjl014** [comment]
Nice work on this, @Mudaafi! Thanks for making the changes! ❤️

**Pull request merged**